presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilterUtils.java

```diff
@@ -347,7 +348,8 @@ public static TupleDomainFilter toBigintValues(long[] values, boolean nullAllowe
         // slots in a hash table), e.g. up to 192 bits per value.
         // Filter based on a bitmap uses (max - min) / num-values bits per value.
         // Choose the filter that uses less bits per value.
-        if ((max - min + 1) > Integer.MAX_VALUE || ((max - min + 1) / values.length) > 192) {
+        BigInteger range = BigInteger.valueOf(max).subtract(BigInteger.valueOf(min)).add(BigInteger.valueOf(1));
```

Comment:
@bhhari nit: perhaps, put each statement on a separate line for readability and consider using intValueExact instead of longValue

```
        BigInteger range = BigInteger.valueOf(max)
                .subtract(BigInteger.valueOf(min))
                .add(BigInteger.valueOf(1));
       if (range.compareTo(BigInteger.valueOf(Integer.MAX_VALUE)) == 1 || (range.intValueExact() / values.length) > 192) {
            return BigintValuesUsingHashTable.of(min, max, values, nullAllowed);
        }
```